components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/autofill/GeckoLoginDelegateWrapper.kt

```diff
@@ -14,18 +17,29 @@ import org.mozilla.geckoview.LoginStorage
  * them to [storageDelegate]. This allows us to avoid duplicating [LoginStorageDelegate] code
  * between different versions of GeckoView, by duplicating this wrapper instead.
  */
-class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) : LoginStorage.Delegate {
+class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) :
+    LoginStorage.Delegate {
+    var job: Job? = null
+
     override fun onLoginSave(login: LoginStorage.LoginEntry) {
         storageDelegate.onLoginSave(login.toLogin())
     }
 
     override fun onLoginFetch(domain: String): GeckoResult<Array<LoginStorage.LoginEntry>>? {
-        val storedLogins = storageDelegate.onLoginFetch(domain)
+        val result = GeckoResult<Array<LoginStorage.LoginEntry>>()
+
+        job?.cancel()
```

Comment:
None of the calls in the `launch` block suspend, so we only cancel if this coroutine has not actually been launched yet. Is that a problem?

In theory if we cancel then the `GeckoResult` we return here will never complete.. Is this a problem? Should we catch a cancellation inside the launch block and make the `GeckoResult` complete? I am not sure if GeckoView expects us to eventually complete it?